ng-intro/ng-introjs.service.ts

```diff
@@ -1,182 +1,182 @@
-	}
-	onHintsAdded(cb:Function){
-		return this.intro.onhintclose(()=> {
-			if(this.isFunction(cb)) cb();
-		});
-	}
+  addListener(name: introStatus, cb: Function) {
+
+    if (this.isFunction(cb))
+      this.notifyList[name] = cb;
+  }
+  //remove from this.notifyList.
+  removeListener(name: introStatus) {
+    delete this.notifyList[name];
+  }
+
+  ///iterate through this.notifyList and call each callback.
+  private notifyListeners(status: introStatus) {
+    for (var key in this.notifyList) {
+      if (this.notifyList.hasOwnProperty(key)) {
+        if (this.isFunction(this.notifyList[key]))
+          this.notifyList[key](status);
+      }
+    }
+  }
+
+  setOptions(options: IntroJs.Options) {
+    return this.intro.setOptions(options);
+  }
+
+  start(step?: number) {
+    if (typeof (step) === 'number') {
+      this.intro.start().goToStep(step);
+    } else {
+      this.intro.start();
+    }
+    this.notifyListeners(introStatus.open);
+
+    return this.intro;
+  }
+
+  exit() {
+    this.notifyListeners(introStatus.closed);
+    return this.intro.exit();
+  }
+
+  clear(cb: Function) {
+    if (typeof (this.intro) !== 'undefined')
+      this.intro.exit();
+
+    this.intro = introJs();
+
+    this.notifyListeners(introStatus.closed);
+
+    if (this.isFunction(cb)) cb();
+
+    return this.intro;
+  }
+
+  addHints() {
+    return this.intro.addHints();
+  }
+  showHint(hintIndex: number) {
+    return this.intro.showHint(hintIndex);
+  }
+  showHints() {
+    return this.intro.showHints();
+  }
+
+  hideHint(hintIndex: number) {
+    return this.intro.hideHint(hintIndex);
+  }
+
+  hideHints() {
+    return this.intro.hideHints();
+  }
+
+  previous() {
+    this.notifyListeners(introStatus.open);
+    return this.intro.previousStep();
+  }
+  next() {
+    this.notifyListeners(introStatus.open);
+    return this.intro.nextStep();
+
+  }
+
+  refresh() {
+    return this.intro.refresh();
+  }
+
+  onComplete(cb: Function) {
+    return this.intro.oncomplete(() => {
+      if (this.isFunction(cb)) cb();
+      this.notifyListeners(introStatus.closed);
+    });
+  }
+  onExit(cb: Function) {
+    return this.intro.onexit(() => {
+      this.notifyListeners(introStatus.closed);
+      if (this.isFunction(cb)) cb();
+    });
+  }
+  onBeforeChange(cb: Function) {
+    return this.intro.onbeforechange(() => {
+      if (this.isFunction(cb)) cb();
+    });
+  }
+  onChange(cb: Function) {
+    return this.intro.onchange(() => {
+      if (this.isFunction(cb)) cb();
+    });
+  }
+  onAfterChange(cb: Function) {
+    return this.intro.onafterchange(() => {
+      if (this.isFunction(cb)) cb();
+    });
+  }
+
+  onHintClick(cb: Function) {
+    return this.intro.onhintclick(() => {
+      if (this.isFunction(cb)) cb();
+    });
+  }
+
+  onHintClose(cb: Function) {
+    return this.intro.onhintclose(() => {
+      if (this.isFunction(cb)) cb();
+    });
+  }
+  onHintsAdded(cb: Function) {
+    return this.intro.onhintclose(() => {
```

Comment:
I've copied your code into my project, but notice this typo that's not your fault, but easily overlooked.  This should use `onhintsadded` here since the above method also calls `onhintclosed`